HotelBooking/HotelBooking.BusinessLogic/Services/Implementation/ReservationService.cs

```diff
@@ -0,0 +1,35 @@
+﻿using HotelBooking.BusinessLogic.Dtos;
+using HotelBooking.BusinessLogic.Services.Abstraction;
+using HotelBooking.DataAccessLayer.Repositories.Interfaces;
+
+namespace HotelBooking.BusinessLogic.Services.Implementation;
+public class ReservationService : IReservationService
+{
+    private IReservationRepository _iReservationRepository;
+
+    public ReservationService(IReservationRepository iReservationRepository)
+    {
+        _iReservationRepository = iReservationRepository;
+    }
+
+    public async Task<ReservationDto> GetReservation(int id)
+    {
+        var reservation = await _iReservationRepository.Get(id);
+        var roomInfos = new List<BasicRoomInfo>();
+
+        roomInfos = reservation.Rooms.Select(c => new BasicRoomInfo(c.Type.ToString(), c.Capacity)).ToList();
+
+        var dto = new ReservationDto()
+        {
+            Id = reservation.ReservationId,
+            DateFrom = reservation.DateFrom,
+            DateTo = reservation.DateTo,
+            Price = reservation.Price,
+            GuestFirstname = await _iReservationRepository.GetGuestFirstname(reservation),
```

Comment:
Jak pisałem niżej te metody GetGuestFirstName i GetGuestLastName nie mogą być w repozytorium. Jeśli chcesz w ten sposób to można na przykład dodać te metody jako prywatne w tej klasie czyli w ReservationService. Będą one pełnić rolę takich metod pomocniczych do wyciągnięcia konkretnej właściwości. Można też po prostu napisać reservation.Guest.LastName i reservation.Guest.FirstName.